in-progress/8757-fees/design.md

```diff
@@ -0,0 +1,265 @@
+| Approvers            | @Maddiaa0 @PhilWindle @dbanks12 @nventuro @joeandrews @aminsammara |
+| Target Approval Date | 2024-10-24                                                         |
+
+## Mana and Costs
+
+Refer to the unit of work on L2 as "mana", analogous to gas on L1. This has been referred to as "L2 gas" in the past.
+This is also separate from the notion of "fee juice", which is conceptually a token with a balance, analogous to ETH.
+
+In this document, we will use the term "fee asset" to refer to the unit of account for fees; that is, a transaction accumulates a cost in mana, which is converted to the fee asset according to the current price of mana.
+
+A transaction incurs the following costs:
+1. Proposers/validators simulate the transaction
+2. Proposers publish pending blocks to L1
+3. Proposers/validators/nodes update their state
+4. Provers generate the avm proofs for the transaction
+5. Provers generate the rollup proofs for the blocks/epoch
+6. Provers verify the epoch proof on L1
+
+Some costs are independent of the computation or data of the transaction itself, specifically numbers 2, 4, and 5; regardless, they must be covered by the transaction.
+
+## Protocol Defined Constants
+
+- `OVERHEAD_MANA_PER_TX` - the overhead cost in mana for a transaction (e.g. 21_000)
+- `TARGET_MANA_PER_BLOCK` - the amount of mana that an "average" block is expected to consume (e.g. 15e6)
+- `BLOBS_PER_BLOCK` - the number of blobs per block that are compensated for in the base fee (e.g. 3)
+- `L2_SLOTS_PER_L2_EPOCH` - the number of L2 slots in each L2 epoch (e.g. 32)
+- `L1_GAS_PER_BLOCK_PROPOSED` - the amount of L1 gas required to propose an L2 block on L1 (e.g. 0.2e6)
+- `L1_GAS_PER_EPOCH_VERIFIED` - the amount of L1 gas required to verify an L2 epoch on L1 (e.g. 1e6)
+- `MINIMUM_PROVING_COST_PER_MANA` - the minimum cost in wei for proving a unit of mana (e.g. 100)
+- `MAXIMUM_PROVING_COST_PERCENT_CHANGE_PER_BLOCK` - the maximum percentage increase in the cost of proving a unit of mana per block (e.g. 1%)
+- `PROVING_COST_PRECISION` - the precision of the `proving_cost_modifier` (e.g. 1e9)
+- `MINIMUM_FEE_ASSET_PRICE` - the minimum price in wei of the fee asset (e.g. TST)
+- `MAXIMUM_FEE_ASSET_PRICE_PERCENT_CHANGE_PER_BLOCK` - the maximum percentage increase in the price of the fee asset per block (e.g. 1%)
+- `FEE_ASSET_PRICE_PRECISION` - the precision of the `fee_asset_price_modifier` (e.g. 1e9)
+- `UNDERLYING_BASE_FEE_ORACLE_UPDATE_INTERVAL` - the minimum number of slots between updates to the underlying base fee oracle (e.g. 4)
+- `MINIMUM_CONGESTION_MULTIPLIER` - the minimum value the congestion multiplier can take (e.g. 1)
+- `CONGESTION_MULTIPLIER_DAMPER` - the constant factor to dampen movement in the congestion multiplier (e.g. 8.547)
+- `MAXIMUM_EPOCH_PROOF_QUOTE_FEE` - the maximum basis point fee a prover can submit for an epoch proof quote (e.g. 9000)
+
+## Block Header Fields
```

Comment:
Not clear to me from this if this is our real block header? Like, we don't need the excess ready for the circuits for example, and we kinda can get away with having for `proven`, `pending` and the last block in each `epoch`. The `base_fee_asset_per_mana` is something we already got as part of the `GasFee` of the global variables (don't recall exact name, think it is `l2_gas` or something like that).